frontend/packages/ceph-storage-plugin/src/components/ocs-install/ocs-request-data.ts

```diff
@@ -0,0 +1,48 @@
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { NO_PROVISIONER } from '../../constants';
+
+export const getOCSRequestData = (
+  scName?: string,
+  storage?: string,
+  infra?: string,
+): K8sResourceKind => {
+  const requestData = {
+    apiVersion: 'ocs.openshift.io/v1',
+    kind: 'StorageCluster',
+    metadata: {
+      name: 'ocs-storagecluster',
+      namespace: 'openshift-storage',
+    },
+    spec: {
+      manageNodes: false,
+      storageDeviceSets: [
+        {
+          name: 'ocs-deviceset',
+          count: 1,
+          replica: 3,
+          resources: {},
+          placement: {},
+          portable: true,
+          dataPVCTemplate: {
+            spec: {
+              storageClassName: scName,
+              accessModes: ['ReadWriteOnce'],
+              volumeMode: 'Block',
+              resources: {
+                requests: {
+                  storage,
+                },
+              },
+            },
+          },
+        },
```

Comment:
nit: If you are refactoring this, can you create one function for deviceSet which will help us in expanding storage by multiple SC in the future.

Reply:
I will prefer to do that as a part of enhancements PR. Since it needs changes in add-capacity modal and requires its testing.

Reply:
Tagging @afreen23. ^^

Reply:
Sounds good @gnehapk . @cloudbehl I can take care of your comment in my PR.